### PR TITLE
Touch up #3559 based on post-merge discussions.

### PR DESCRIPTION
**Pull request by jmchilton**
xref #3559

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/folder_contents.py [hunk 1]
```diff
@@ -271,22 +271,22 @@ def create( self, trans, encoded_folder_id, payload, **kwd ):
...
         try:
             if from_hda_id:
                 decoded_hda_id = self.decode_id( from_hda_id )
-                rvals.append(self._copy_hda_to_library_folder( trans, self.hda_manager, decoded_hda_id, encoded_folder_id_16, ldda_message ))
+                return self._copy_hda_to_library_folder( trans, self.hda_manager, decoded_hda_id, encoded_folder_id_16, ldda_message )
             if from_hdca_id:
+                rvals = []
                 decoded_hdca_id = self.decode_id( from_hdca_id )
                 rvals.extend(self._copy_hdca_to_library_folder( trans, self.hda_manager, decoded_hdca_id, encoded_folder_id_16, ldda_message ))
+                return rvals
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: You can remove `rvals` and just do:
```python
                 return self._copy_hdca_to_library_folder( trans, self.hda_manager, decoded_hdca_id, encoded_folder_id_16, ldda_message )
```

**Review by nsoranzo — CHANGES_REQUESTED** [review]
`:returns:` and `:rtype:` of the 2 methods' documentation need to be updated accordingly.

**Comment by jmchilton** [comment]
@nsoranzo I think I rebased this to address your concerns - let me know if this is not the case.

File: lib/galaxy/webapps/galaxy/api/folder_contents.py [hunk 2]
```diff
@@ -258,8 +258,9 @@ def create( self, trans, encoded_folder_id, payload, **kwd ):
             :type  extended_metadata:   dict
         :type   payload:    dict
 
-        :returns:   a list of dictionaries containing the id, name, and 'show' url of the new item
-        :rtype:     list
+        :returns:   a dictionary describing the new item if ``from_hda_id`` is supplied or a list of
+                    of dictionaries describing the new items if ``from_hdca_id`` is supplied.
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: Double `of` ;)

File: lib/galaxy/webapps/galaxy/api/folder_contents.py [hunk 3]
```diff
@@ -271,22 +272,20 @@ def create( self, trans, encoded_folder_id, payload, **kwd ):
...
-        rvals = []
         try:
             if from_hda_id:
                 decoded_hda_id = self.decode_id( from_hda_id )
-                rvals.append(self._copy_hda_to_library_folder( trans, self.hda_manager, decoded_hda_id, encoded_folder_id_16, ldda_message ))
+                return self._copy_hda_to_library_folder( trans, self.hda_manager, decoded_hda_id, encoded_folder_id_16, ldda_message )
             if from_hdca_id:
                 decoded_hdca_id = self.decode_id( from_hdca_id )
-                rvals.extend(self._copy_hdca_to_library_folder( trans, self.hda_manager, decoded_hdca_id, encoded_folder_id_16, ldda_message ))
+                return self._copy_hdca_to_library_folder( trans, self.hda_manager, decoded_hdca_id, encoded_folder_id_16, ldda_message ))
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/folder_contents.py]: There's an extra `)` at the end here.

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
@mvdbeek Merge?

**Review by mvdbeek — APPROVED** [review]
Yes, that looks better. Thanks @jmchilton and @nsoranzo !

**Comment by nsoranzo** [comment]
Thanks, BioBlend is fixed on `dev` now!

**Pull request merged**